scanner/src/main/kotlin/provenance/NestedProvenanceScanResult.kt

```diff
@@ -130,6 +133,27 @@ data class NestedProvenanceScanResult(
         return findings
     }
 
+    private fun Map<KnownProvenance, List<ScanResult>>.mergeSnippetFindings(): Set<SnippetFinding> {
+        val findingsByPath = mapKeys { getPath(it.key) }.mapValues { (_, scanResults) ->
+            mutableSetOf<SnippetFinding>().also { acc ->
+                scanResults.forEach { acc += it.summary.snippetFindings }
+            }
+        }
+
+        val allFindings = mutableSetOf<SnippetFinding>()
+
+        findingsByPath.map { (path, findings) ->
```

Comment:
Use `forEach` instead as you don't use the result. You could also use `buildSet`:
```kotlin
return buildSet {
  findingsByPath.forEach { (path, findings) ->
    ...
    addAll(findings.map { ... })
}
```